### PR TITLE
New Post Display Mode Picker

### PR DESCRIPTION
**Pull request by Sjmarf**
Implemented a nicer post display mode picker in Appearance Settings. 

Previously I also had a preview section for the other customisation options planned. I'll be waiting until 1.2 before proceeding with that one, as we may decide to implement a drag-and-drop customisation system instead. [Discussion on Slack here](https://mlemhq.slack.com/archives/C05J3BS4FQU/p1689959092754699?thread_ts=1689953046.293379&cid=C05J3BS4FQU).

Screenshot 2023-07-21 at 18 45 01

### CODE REVIEW

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/Post/PostSettingsView.swift [hunk 1]
```diff
@@ -40,13 +40,20 @@ struct PostSettingsView: View {
...
-            Section("Post Size") {
-                SelectableSettingsItem(
-                    settingIconSystemName: "rectangle.compress.vertical",
-                    settingName: "Post size",
-                    currentValue: $postSize,
-                    options: PostSize.allCases
-                )
+            Section {
+                HStack {
+                    PostDisplayModeView(postSize: .large, imageName: "LargePost", selected: $postSize)
```
[the remark this answers is not in the document]
Comment (Sjmarf) [reply Mlem/Views/Tabs/Settings/Components/Views/Appearance/Post/PostSettingsView.swift:45]: Sure, I can do that.

File: Mlem/Assets.xcassets/PostIcons/CompactPost.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "postmode3.svg",
```
[the remark this answers is not in the document]
Comment (WestonHanners) [reply Mlem/Assets.xcassets/PostIcons/CompactPost.imageset/Contents.json:4]: Yes Please! 🙏

**Comment by Sjmarf** [comment]
> where did the SVGs originate from?

I made them. I look into making them load faster.

**Comment by Sjmarf** [comment]
I've made the files as small as possible, but there's still a small amount of lag when opening the page. It might be possible to change the way Swift loads the images, or use PNGs instead?

File: Mlem/Assets.xcassets/PostIcons/CompactPost.imageset/Contents.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sjmarf) [reply Mlem/Assets.xcassets/PostIcons/CompactPost.imageset/Contents.json:4]: Done

**Comment by EricBAndrews** [comment]
I wonder if we'd get better performance if we rendered them in pure SwiftUI instead of loading images

**Comment by Sjmarf** [comment]
> I wonder if we'd get better performance if we rendered them in pure SwiftUI instead of loading images

I'll try it 👍

**Review by WestonHanners — APPROVED** [review]

**Comment by Sjmarf** [comment]
Closing and reimplementing differently in a future PR.

**Pull request closed without merging**